### DISCUSSION TITLE
Difference between headless and not headless when it comes to detection

### DISCUSSION THREAD

**Discussion by AlexPaiva** [ultrafunkamsterdam/undetected-chromedriver#1062] [Q&A] [answered]
So, I have been using non-headless for all my automation scripts to log in into instagram and grab a quick reference to save me time, but obviously I would prefer to do it headless. At the time I head headless was easy to be detected so I never messed with it any more but now with the latest update on UC I noticed that now headless is also undetected, would this be in any way a viable option to use that avoid detection or should I stick with non-headless?

**Comment by ultrafunkamsterdam** [comment, marked as answer]
you are lucky. yesterday i created a new version 3.4.5 which includes a patch for headless, and my tests rendered Undetected both linux and windows.
Viable. yes.. supported : still not atm

seems like google is in for the long run
https://support.google.com/chrome/a/answer/7679408#hdlssMod110 
(open in incognito browser for english version, not all languages contain this piece)

**Reply by AlexPaiva** [reply]
So I can use it 'freely' for now as it will do the exact same as non-headless?

When you say not supported, what do you mean exactly?

**Reply by ultrafunkamsterdam** [reply]
Yes you can. but this only works since yesterday, too few people reported back to make a judgement.
Of course headless can never be exact the same, but at least for now detection-wise it performs good.
When i say not supported that means  if you have trouble, get detected , or want an extension in it or whatever, you must be lucky someone will help you but i won't ❤️

**Reply by AlexPaiva** [reply]
Thanks for the reply, gave it a try and doesn't seem to work for what I have but it does login to a very popular website easily so for some tasks it might just work.
